c/misra/src/rules/DIR-5-3/ThreadCreatedByThread.ql

```diff
@@ -0,0 +1,45 @@
+/**
+ * @id c/misra/thread-created-by-thread
+ * @name DIR-5-3: Threads shall not be created by other threads
+ * @description Creating threads within threads creates uncertainty in program behavior and
+ *              concurrency overhead costs.
+ * @kind problem
+ * @precision high
+ * @problem.severity error
+ * @tags external/misra/id/dir-5-3
+ *       external/misra/c/2012/amendment4
+ *       correctness
+ *       maintainability
+ *       concurrency
+ *       performance
+ *       external/misra/obligation/required
+ */
+
+import cpp
+import codingstandards.c.misra
+import codingstandards.cpp.Concurrency
+
+Function callers(Function f) { result = f.getACallToThisFunction().getEnclosingFunction() }
+
+class ThreadReachableFunction extends Function {
```

Comment:
I think this would be nicer if you rewrote this as a `ThreadRoot` class, that provides accessors `getThreadCreateCall()` and `getAReachableFunction()`.

Then the query could be rewritten as:
```codeql
from ThreadRoot threadRoot, CThreadCreateCall tc
where
  not isExcluded(tc, Concurrency6Package::threadCreatedByThreadQuery()) and
  threadRoot.getAReachableFunction() = tc.getEnclosingFunction()
select tc, "Thread creation call reachable from threaded function '$@' created by call to $@.", threadRoot,
  threadRoot.getName(), threadRoot.getThreadCreateCall() as tcOriginal, tcOriginal.getTarget().getName()
```

Note, I'm also suggesting here tagging in the original `CThreadCreateCall`, so that it's easier for a user to verify that `threadRoot` is actually a threaded function.